ReportingServicesTools/Functions/Admin/Restore-RsEncryptionKey.ps1

```diff
@@ -75,7 +75,8 @@ function Restore-RSEncryptionKey
     {
         $rsWmiObject = New-RsConfigurationSettingObjectHelper -BoundParameters $PSBoundParameters
 
-        $KeyPath = Resolve-Path $KeyPath
+       
+        $KeyPath = Join-Path -Path (Resolve-Path -Path (Split-Path -Parent $KeyPath)).ProviderPath -ChildPath (Split-Path -Leaf $KeyPath)
```

Comment:
instead of this consider doing 

$KeyPath = (Resolve-Path $KeyPath).ProviderPath